### ISSUE TITLE
Can `lax.pow` Improve Performance

### ISSUE THREAD

**Issue by Jordan-Dennis** [LouisDesdoigts/dLux#169]
Hi all,
The exponentiation is a considerable cost of the `basis` generation. We may be able to curb that by incorporating one of `lax.interger_pow` or `lax.pow` instead of using `**` which is treated as `np.power`.

**Comment by Jordan-Dennis** [comment]
So there have been some interesting developments regarding this. Firstly, `lax.integer_pow` is not `jit` comilable. This is not surprising since the source code uses a `while` loop. The algorithm does appear to be faster than `lax.pow` so it may be worth investing some time in making this optimisation. However, looking at the algorithm it does not look like I will be able to make it compile without sacrificing the things that make it fast.

**Comment by Jordan-Dennis** [comment]
Secondly, using `lax.pow` instead of `**` will make a difference I believe because it will kill `convert_dtype` in the `jaxpr`.

**Comment by benjaminpope** [comment]
I don’t think we are especially worried about fast compiled basis generation, and I am surprised if jax and lax elementary arithmetic options make much of a difference?

———————————————
Dr Benjamin Pope (he/him)
Lecturer in Astrophysics
University of Queensland
benjaminpope.github.io
________________________________

**Comment by Jordan-Dennis** [comment]
Yeah so, if you look at the `jaxpr` a lot of the `np` functions do type checking and extra broadcasting (presumably for safety in the general case). If you check issue #178 you'll see me accelerate some operations by avoiding these assumptions.
